registry/slashFun.js

```diff
@@ -0,0 +1,163 @@
+// @ts-check
+const u = require('./regUtils');
+
+const roll = new u.sub()
+  .setName("roll")
+  .setDescription("Roll Dice")
+  .addIntegerOption(
+    new u.int()
+    .setName("dice")
+    .setDescription("How many dice to roll? (will roll at least 1 anyway) (max of 10000)")
+    .setRequired(false)
+    .setMinValue(1)
+    .setMaxValue(10000)
+  )
+  .addIntegerOption(
+    new u.int()
+    .setName("sides")
+    .setDescription("How many sides on the dice to roll? (defaults to 6)")
+    .setRequired(false)
+    .setMinValue(1)
+    .setMaxValue(2147483647)
+  )
+  .addIntegerOption(
+    new u.int()
+    .setName("modifier")
+    .setDescription("How much to change the roll by? (defaults to 0)")
+    .setRequired(false)
+  );
+const rollF = new u.sub()
+  .setName("rollf")
+  .setDescription("Roll Fate Dice")
+  .addIntegerOption(
+    new u.int()
+    .setName("dice")
+    .setDescription("How many dice to roll? (will roll at least 1 anyway) (max of 10000)")
+    .setRequired(false)
+    .setMinValue(1)
+    .setMaxValue(10000)
+  )
+  .addIntegerOption(
+    new u.int()
+    .setName("modifier")
+    .setDescription("How much to change the roll by? (defaults to 0)")
+    .setRequired(false)
+  );
+const rollOld = new u.sub()
+  .setName("rollold")
+  .setDescription("Roll Dice using the old formula format")
+  .addStringOption(
+    new u.string()
+    .setName("rollformula")
+    .setDescription("old dice formula to use, it errors in the same way as !roll did")
+  );
+const ball8 = new u.sub()
+  .setName("8ball")
+  .setDescription("Get an answer from the Magic 8-ball.")
+  .addStringOption(
+    new u.string()
+    .setName("question")
+    .setDescription("What do you wish to ask the 8-ball today?")
+    .setRequired(true)
+  );
+const allthe = new u.sub()
+  .setName("allthe")
+  .setDescription("ALL THE _____!")
+  .addStringOption(
+    new u.string()
+    .setName("thing")
+    .setDescription("something")
+    .setRequired(true)
+  );
```

Comment:
This should be migrated a tag, no slash command or code required.

Reply:
trimmed all this stuff.